src/lib/dialog/dialog-directives.ts

```diff
@@ -0,0 +1,49 @@
+import {Directive} from '@angular/core';
+import {MdDialog} from './dialog';
+
+
+/**
+ * Button that will close the current dialog.
+ */
+@Directive({
+  selector: 'button[md-dialog-close]',
+  host: {
+    '(click)': 'dialog.closeTop()'
```

Comment:
I don't think this will work for situations where there are two dialogs open side-by-side instead of one on top of the other. I took a poke at the code and found that you can inject the `MdDialogRef` here because the content is loaded with an instance of `DialogInjector`, so this works:
```ts
/**
 * Button that will close the current dialog.
 */
@Directive({
  selector: 'button[md-dialog-close]',
  host: {
    '(click)': 'dialogRef.close()'
  }
})
export class MdDialogClose {
  constructor(public dialogRef: MdDialogRef<any>) { }
}
```

With this change, I don't think we need `closeTop` any more.